src/wasm2asm.h

```diff
@@ -522,11 +526,54 @@ void Wasm2AsmBuilder::addTables(Ref ast, Module* wasm) {
 void Wasm2AsmBuilder::addExports(Ref ast, Module* wasm) {
   Ref exports = ValueBuilder::makeObject();
   for (auto& export_ : wasm->exports) {
-    ValueBuilder::appendToObject(
-      exports,
-      fromName(export_->name),
-      ValueBuilder::makeName(fromName(export_->value))
-    );
+    if (export_->kind == ExternalKind::Function) {
+      ValueBuilder::appendToObject(
+        exports,
+        fromName(export_->name),
+        ValueBuilder::makeName(fromName(export_->value))
+      );
+    }
+    if (export_->kind == ExternalKind::Memory) {
+      almostASM = true;
```

Comment:
I don't think this should depend on the presence of a memory export - maybe I'm missing something?

I think it should happen of growth is used, as I see later in the patch, and also the memory has an initial and a max value; if the max is larger than the initial, then growth is possible (from outside - maybe we don't care about that?).

Reply:
I don't think asm.js exports can be anything other than functions.

http://asmjs.org/spec/latest/#validateexport-s

Reply:
Is enforcing the maximum growth important for a first pass?  The primary motivation is getting backward compatibility when targeting wasm first.

Reply:
I see what you mean now, yeah, asm.js doesn't allow exporting the memory. However, I think we should just not export the memory - asm.js always imports (and never exports) the memory, no matter what wasm does, so that is always going to be different.

About enforcing the maximum growth, I agree, we can leave that for later.

Reply:
The export shims the WebAssembly.Memory interface and is also the only way to get the reassigned buffer post memory growth, the imported buffer is only the initial buffer.

Reply:
I see, thanks, then I agree we need to export it and disable "use asm".

However, can we not do this if memory growth is disabled? I think it may be common to have a non-memory growth wasm module that exports a memory, and it would be better to not export it so that we can keep "use asm" for speed. But, a downside of that is we do something different depending on memory growth. Perhaps we can issue a warning, and/or have an option to ignore a memory export?

Reply:
Exporting memory allows for memory growth externally, via the grow function even if the module itself doesn't have grow_memory instructions.  I will log a message when "almost asm" is triggered with the reason why.

Reply:
I'm excited for this!

Reply:
Ok, this looks great! All ready to merge I think, except for the logging just mentioned, which I don't think I see yet?